tasks/create-internal-request/create-internal-request.yaml

```diff
@@ -53,6 +53,46 @@ spec:
     - name: input
       description: Workspace to store the params and responses for the internalRequest
   steps:
+    - name: merge-internal-request-parameters
+      image:
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      script: |
+          #!/usr/bin/env bash
+          DATA_FILE="$(workspaces.input.path)/$(params.dataPath)"
+          if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+          fi
+
+          IR_PARAMS_FILE="$(workspaces.input.path)/$(params.subdirectory)/ir-params-$$.txt"
+          cat > ${IR_PARAMS_FILE} <<JSON
+          $(params.internalRequestParams)
+          JSON
+
+          # to keep the compatibility the internalrequest parameters must be merged
+          # with the data json
+          JSON_PARAMS="$(jq -c '.' ${IR_PARAMS_FILE})"
+          INDEX=$(jq '.|length' ${IR_PARAMS_FILE})
+          jq '.fbc' $(workspaces.input.path)/$(params.dataPath) |yq -oy -P |while read PARAM_VALUE; do
```

Comment:
Can we iterate over indexes here? It seems wrong and fragile to manually get the key:value pairs from yaml output and there's not need for it. We can easily just get the key and value separately in each iteration and use that.

Reply:
It did not work. I've done it like this due to the format of the data, which in this case is:
```
paramA:  valueA
paramB: valueB
paramC: valueC
```

Reply:
Ok, saw that I can use `to_entries` to do that.

Reply:
rewrote.